components/service/glean/src/test/java/mozilla/components/service/glean/FakeDispatchersInTest.kt

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import org.junit.AfterClass
+import org.junit.BeforeClass
+
+/**
+ * A class that stubs the coroutine scopes in [Dispatchers] to allow blocking
+ * on async API calls when running tests.
+ *
+ * Note: according to [kotlinx.coroutines.Dispatchers.Unconfined], this will work
+ * as long as the API doesn't use any suspending function.
+ */
+@ExperimentalCoroutinesApi @ObsoleteCoroutinesApi
+open class FakeDispatchersInTest {
```

Comment:
A more idiomatic way would be to write this as a JUnit `Rule` that you can apply to tests:

* https://github.com/junit-team/junit4/wiki/rules
* https://medium.com/@vanniktech/writing-your-own-junit-rule-3df41997b10c